src/plugins/data/common/opensearch_query/filters/geo_shape_filter.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoShapeRelation } from '@opensearch-project/opensearch/api/types';
+import { Filter, FilterMeta } from './meta_filter';
+
+export type Position = number[];
+
+export interface PreIndexedShapeFilter {
+  index: string;
+  id: string;
+  path: string;
+  routing?: string;
+}
+
+export interface Polygon {
+  type: 'Polygon';
+  coordinates: Position[][];
+}
+
+export interface MultiPolygon {
+  type: 'MultiPolygon';
+  coordinates: Position[][][];
+}
+
+// Will support polygon & multi polygon at this moment
```

Comment:
Do we need a "ToDo" to potentially add others?

Reply:
Sure. Does adding TODO is enough or do you recommend github issue too?

Reply:
You know what? GH issue would be the best :)

Reply:
Added https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3612